### PR TITLE
Ported over the Json{Reader, Writer, Navigator} Tests

### PR DESCRIPTION
**Pull request by bchong95**
Ported over the Json{Reader, Writer, Navigator} Tests

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.NetFramework.Tests/Json/JsonMicroBenchmarks.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+namespace Microsoft.Azure.Cosmos.NetFramework.Tests.Json
+{
+    using System.Collections.Generic;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class JsonMicroBenchmarks
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.NetFramework.Tests/Json/JsonMicroBenchmarks.cs]: Do we need BenchMark really in CI tests? 
If not may be performance test project mgiht be right place. Please check with Matias.

Comment (bchong95) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.NetFramework.Tests/Json/JsonMicroBenchmarks.cs]: It doesn't need to be here. I was mainly just doing a blind port.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.NetFramework.Tests/Json/JsonNewtonsoftCosmosDBBinaryReader.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//-----------------------------------------------------------------------
+namespace Microsoft.Azure.Cosmos.NetFramework.Tests.Json
+{
+    using System.IO;
+    using Microsoft.Azure.Cosmos.Json;
+
+    /// <summary>
+    /// Wrapper class that implements an IJsonReader that really is a wrapper around our Newtonsoft wrapper for binary.
+    /// </summary>
+    internal sealed class JsonNewtonsoftCosmosDBBinaryReader : JsonNewtonsoftReader
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.NetFramework.Tests/Json/JsonNewtonsoftCosmosDBBinaryReader.cs]: Isn't it true that the implementations will be in soruce eventaully. 
Respecrtive tests only test the source implementation aspects. Isn't it?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.NetFramework.Tests/Json/JsonWriterTests.cs [hunk 3]
```diff
@@ -0,0 +1,1285 @@
...
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Microsoft.Azure.Cosmos.Json;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System.Reflection;
+
+    [TestClass]
+    public class JsonWriterTests
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.NetFramework.Tests/Json/JsonWriterTests.cs]: Please move these into Comsos.Tests.

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.NetFramework.Tests/Json/JsonWriterTests.cs]: In general all simple UT belongs to CosmosTests.

**Pull request merged**